base/src/main/java/com/nedap/archie/base/Interval.java

```diff
@@ -32,43 +31,51 @@
     /**
      * lower boundary open (i.e. = -infinity)
      */
-    @XmlAttribute(name="lower_unbounded")
+	@XmlAttribute(name = "lower_unbounded")
     private boolean lowerUnbounded = false;
     /**
      * upper boundary open (i.e. = +infinity)
      */
-    @XmlAttribute(name="upper_unbounded")
+	@XmlAttribute(name = "upper_unbounded")
     private boolean upperUnbounded = false;
     /**
      * lower boundary value included in range if not lower_unbounded.
      */
-    @XmlAttribute(name="lower_included")
+	@XmlAttribute(name = "lower_included")
     private boolean lowerIncluded = true;
     /**
      * upper boundary value included in range if not upper_unbounded.
      */
-    @XmlAttribute(name="upper_included")
+	@XmlAttribute(name = "upper_included")
     private boolean upperIncluded = true;
 
     public Interval() {
 
     }
 
     public Interval(T value) {
-        this.lower = value;
-        this.upper = value;
+		this(value, value);
     }
 
     public Interval(T lower, T upper) {
-        this.lower = lower;
-        this.upper = upper;
+		this(lower, upper, true, true);
     }
 
     public Interval(T lower, T upper, boolean lowerIncluded, boolean upperIncluded) {
         this.lower = lower;
         this.upper = upper;
         this.lowerIncluded = lowerIncluded;
         this.upperIncluded = upperIncluded;
+
+		if (upper == null) {
+			this.upperUnbounded = true;
+			this.upperIncluded = true;
```

Comment:
Invariant from RM spec:
```
Lower_included_valid: lower_unbounded implies not lower_included
Upper_included_valid: upper_unbounded implies not upper_included
```

Not that it matters much, if you set lower_unbounded lower_included stops to mean anything, but according to the specs in this case lowerIncluded/upperIncluded should be false :)

We could use an invariant validator I guess :)

Reply:
This is the same as in behavior
public static <T> Interval<T> lowerUnbounded(T upper, boolean upperIncluded) 

and the tests depend on that.